### PR TITLE
feat: add contributor page

### PR DESCRIPTION
**Pull request by josefaidt**
#### Description of changes:

- adds "contributor" page from https://www.contributetoamplify.com/
- adds `/contribute`
- adds `/contribute/getting-started`
- ports changes to docs Layout

Todo:

- [x] fix layout with quickstart videos, ensure video layout is sufficient
- [ ] add a link to this page on the landing page(?)
- [x] refactor page metadata out to utility file

Question:
- [ ] Is anything needing to make sure that the page is picked up by the site search index?

image (1)

image (2)

image (3)

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] iOS
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [x] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?
- [x] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.
- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created? 
- [x] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br />
      _ref: MDX: `[link](https://link.com)`
            HTML: `<a href="https://link.com">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/pages/contribute/index.tsx [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+        <meta property="og:locale" content="en-us" />
+
+        <meta name="twitter:card" content="summary_large_image" />
+        <meta name="twitter:site" content={creator} />
+        <meta name="twitter:creator" content={creator} />
+        <meta name="twitter:title" content={meta.title} />
+        <meta name="twitter:description" content={meta.description} />
+        {/* <meta
+          property="twitter:image"
+          content="https://contributetoamplify.com/contributor-bash-og-image.png"
```
Comment (josefaidt) [inline src/pages/contribute/index.tsx:83]: @todo add proper/correct opengraph image

File: src/pages/contribute/index.tsx [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+                  title={'Amplify CLI'}
+                  issues={CLIissues}
+                  repo={'amplify-cli'}
+                  description="The AWS Amplify CLI is a toolchain for simplifying serverless web and mobile development."
+                />
+              </Card>
+            </Flex>
+
+            <QuickstartResources />
+            {/* <Contributors /> */}
```
Comment (katiegoines) [inline src/pages/contribute/index.tsx:181]: Are we not including this component anymore?

File: src/pages/contribute/index.tsx (hunk 1 above)
Comment (siegerts) [reply src/pages/contribute/index.tsx]: @josefaidt this gtg?

File: src/pages/contribute/index.tsx (hunk 2 above)
Comment (siegerts) [reply src/pages/contribute/index.tsx:181]: We'll open an issue for this and follow up.

**Review by katiegoines — DISMISSED** [review]

**Review by offlineprogrammer — APPROVED** [review]

File: src/pages/contribute/index.tsx (hunk 1 above)
Comment (josefaidt) [reply src/pages/contribute/index.tsx]: yep, it's been updated

```
<meta
          property="twitter:image"
          content="/images/contribute/contributor-bash-og-image.png"
        />
```

**Review by siegerts — DISMISSED** [review]

**Review by offlineprogrammer — APPROVED** [review]

**Review by siegerts — DISMISSED** [review]
LGTM!

**Review by offlineprogrammer — DISMISSED** [review]

**Review by katiegoines — APPROVED** [review]

**Review by mauerbac — APPROVED** [review]

**Pull request merged**